contracts/src/assertionStakingPool/AssertionStakingPool.sol

```diff
@@ -0,0 +1,97 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+// SPDX-License-Identifier: BUSL-1.1
+//
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../rollup/IRollupLogic.sol";
+import "../rollup/IRollupCore.sol";
+import "./StakingPoolErrors.sol";
+
+contract AssertionStakingPoolCreator {
+    function createPoolForAssertion(address _rollup, bytes32 _expectedAssertionHash) external {
+        new AssertionStakingPool(_rollup, _expectedAssertionHash);
+    }
+}
+
+contract AssertionStakingPool {
+    using SafeERC20 for IERC20;
+
+    address public immutable rollup;
+    bytes32 public immutable assertionHash;
+    IERC20 public immutable stakeToken;
+    uint256 public immutable baseStake;
+
+    mapping(address => uint256) public depositedTokenBalances;
+
+    PoolState public poolState = PoolState.PENDING;
+
+    constructor(address _rollup, bytes32 _assertionHash) {
+        rollup = _rollup;
+        assertionHash = _assertionHash;
+        stakeToken = IERC20(IRollupCore(rollup).stakeToken());
+        baseStake = IRollupCore(rollup).baseStake();
+
+        stakeToken.approve(rollup, baseStake);
```

Comment:
if the base stake changes the baseStake value would be reset thought, right? Since the constructor is already committing to a specific assertion, isn't fetching it from the rollup safe / current?